src/main/java/randoop/main/Minimize.java

```diff
@@ -0,0 +1,1185 @@
+   *   <li>Remove the left hand side of a statement, retaining only the expression on the right.
+   * </ul>
+   *
+   * <p>Assertions are never simplified, only removed completely.
+   *
+   * @param currStmt statement to simplify
+   * @param primitiveValues map of primitive variable names to expressions representing their values
+   * @return list of statements, where each is a possible simplification of {@code currStmt}
+   */
+  private static List<Statement> getStatementReplacements(
+      Statement currStmt, Map<String, String> primitiveValues) {
+    List<Statement> replacements = new ArrayList<Statement>();
+
+    // Null represents removal of the statement.
+    replacements.add(null);
+
+    if (currStmt instanceof ExpressionStmt) {
+      Expression exp = ((ExpressionStmt) currStmt).getExpression();
+      if (exp instanceof VariableDeclarationExpr) {
+        // Create and return a list of possible replacement statements.
+        VariableDeclarationExpr vdExpr = (VariableDeclarationExpr) exp;
+
+        // Simplify right hand side to zero-equivalent value:  0, false, or null.
+        Statement rhsZeroValStmt = rhsAssignZeroValue(vdExpr);
+        if (rhsZeroValStmt != null) {
+          replacements.add(rhsZeroValStmt);
+        }
+
+        // Simplify right hand side to a value that was previously found
+        // in a passing assertion.
+        Statement rhsAssertValStmt = rhsAssignValueFromPassingAssertion(vdExpr, primitiveValues);
+        if (rhsAssertValStmt != null) {
+          replacements.add(rhsAssertValStmt);
+        }
+
+        // Simplify statement by removing the left hand side.
+        Statement lhsRemovalStmt = removeLeftHandSideSimplification(vdExpr);
+        if (lhsRemovalStmt != null) {
+          replacements.add(lhsRemovalStmt);
+        }
+      }
+    }
+    return replacements;
+  }
+
+  /**
+   * If {@code currStmt} is a statement that is an assertTrue statement using an '==' operator,
+   * store the value associated with the variable in the {@code primitiveValues} map.
+   *
+   * @param currStmt a statement
+   * @param primitiveValues a map of variable names to variable values; modified if {@code currStmt}
+   *     is a passing assertion, asserting a variable's value.
+   */
+  private static void storeValueFromAssertion(
+      Statement currStmt, Map<String, String> primitiveValues) {
+    // Check if the statement is an assertion regarding a value that can be
+    // used in a simplification later on.
+    if (currStmt instanceof ExpressionStmt) {
+      Expression exp = ((ExpressionStmt) currStmt).getExpression();
+      if (exp instanceof MethodCallExpr) {
+        MethodCallExpr mCall = (MethodCallExpr) exp;
+        // Check if the method call is an assertTrue statement
+        if (mCall.getName().equals("assertTrue")) {
+          // Is an assertTrue statement.
+          List<Expression> mArgs = mCall.getArgs();
+          if (mArgs.size() == 1) {
```

Comment:
This does not accommodate the [two-argument version of assertTrue](http://junit.sourceforge.net/javadoc/org/junit/Assert.html).